src/main/java/org/jenkinsci/plugins/displayurlapi/DisplayURLContext.java

```diff
@@ -95,39 +109,61 @@ private void guessPlugin() {
             if (ourPluginClassNames.contains(cname)) {
                 continue;
             }
-            try {
-                Class<?> clazz = loader.loadClass(cname);
-                PluginWrapper wrapper = manager.whichPlugin(clazz);
-                if (wrapper != null && !OUR_PLUGIN_NAME.equals(wrapper.getShortName())) {
-                    plugin = wrapper;
-                    break;
+            Optional<PluginWrapper> wrapper = CACHE.getIfPresent( cname);
+            if(wrapper != null){
+                if(wrapper.isPresent()){
+                    plugin = wrapper.get();
+                }
+            } else
+            {
+                try {
+                    Class<?> clazz = loader.loadClass( cname );
+                    PluginWrapper pluginWrapper = manager.whichPlugin( clazz );
+                    if ( pluginWrapper != null && !OUR_PLUGIN_NAME.equals( pluginWrapper.getShortName() ) ) {
+                        plugin = pluginWrapper;
+                        CACHE.put( cname, Optional.of( pluginWrapper ) );
+                        break;
+                    } else {
+                        CACHE.put( cname, Optional.empty() );
+                    }
+                } catch ( ClassNotFoundException e ) {
+                    // ignore, it's not a plugin
                 }
-            } catch (ClassNotFoundException e) {
-                // ignore, it's not a plugin
             }
         }
     }
 
+
     /**
      * Opens a {@link DisplayURLContext} for the current thread.
-     *
+     * @param guessPlugin to try and infer the current plugin (a bit resources consuming as inspect stacktrace and classloader)
```

Comment:
nit: maybe worth expanding on the comment like this:

```suggestion
     * @param guessPlugin try to infer the current plugin (resource intensive as requires a stack trace and class loading). Use {@code false} if you know the caller will always be Jenkins core.
```